shared/src/token_info.rs

```diff
@@ -0,0 +1,81 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use contracts::ERC20;
+use ethcontract::{Http, Web3, H160};
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+use mockall::*;
+
+#[cfg_attr(test, derive(Eq, PartialEq))]
+#[derive(Copy, Clone, Debug)]
+pub struct TokenInfo {
+    pub decimals: u8,
+}
+
+pub struct TokenInfoFetcher {
+    //pub factory: UniswapV2Factory,
+    pub web3: Web3<Http>,
+    //pub chain_id: u64,
+}
+
+#[automock]
```

Comment:
nit: this can be made "test-only" by using:

```rust
#[cfg_attr(test, mockall::automock)]
```

(cf. e.g. `BalanceFetching`)

Reply:
Problem is that it is in a separate crate so when the `solver` is compiled in test, it's dependency `shared` will not be in test. Iirc there is an outstanding cargo issue/feature to allow using the same dependency also as dev-depencency (used by tests) with different features.